src/mlpack/methods/ann/layer/lookup_impl.hpp

```diff
@@ -54,17 +63,34 @@ void Lookup<InputDataType, OutputDataType>::Gradient(
     const arma::Mat<eT>& error,
     arma::Mat<eT>& gradient)
 {
-  gradient = arma::zeros<arma::Mat<eT> >(weights.n_rows, weights.n_cols);
-  gradient.cols(arma::conv_to<arma::uvec>::from(input) - 1) = error;
+  const size_t seqLength = input.n_rows;
+  const size_t batchSize = input.n_cols;
+
+  arma::Cube<eT> errorTemp(const_cast<arma::Mat<eT>&>(error).memptr(),
+      embeddingSize, seqLength, batchSize, true, false);
```

Comment:
Hey there, Since errorTemp is not modified anywhere, we should avoid creating a copy. See Convolutional Layers, Transposed Convolution Layers for reference. Thanks a lot.

Reply:
Yes. Sure, I will correct it in the next commit. :+1:

Reply:
Awesome, Thanks a lot.